ansible_rulebook/util.py

```diff
@@ -208,3 +212,27 @@ def check_jvm():
             file=sys.stderr,
         )
         sys.exit(1)
+
+
+def has_builtin_filter(name: str) -> bool:
+    return _builtin_filter_path(name)[0]
+
+
+def find_builtin_filter(name: str) -> str:
+    found, path = _builtin_filter_path(name)
+    if found:
+        return path
+    return None
+
+
+def _builtin_filter_path(name: str) -> Tuple[bool, str]:
+    if not name.startswith(EDA_BUILTIN_FILTER_PREFIX):
+        return False, ""
+    filter_name = name.split(".")[-1]
+
+    if not filter_name:
+        raise InvalidFilterNameException(name)
+
+    dirname = os.path.dirname(os.path.realpath(__file__))
+    path = os.path.join(dirname, "event_filters", filter_name + ".py")
+    return os.path.exists(path), path
```

Comment:
The logic of these three functions seems a bit odd for me. Returning a boolean with the result seems more a pattern of golang rather than python. I feel it could be simpler if `has_builtin_filter` just check if the source has a valid builtin filter name and `find_builtin_filter` (which seems to be more like `get_builtin_filter`) returns an exception if the source filter doesn't exist.

Reply:
I mean something like:
```python
def has_builtin_filter(name: str) -> bool:
    return name.startswith(EDA_BUILTIN_FILTER_PREFIX):
    
def get_builtin_filter_path(name: str) -> str:
    if not has_builtin_filter(name):
        raise InvalidFilterNameException(f"Invalid builtin filter name: {name}")
    filter_name = name.split(".")[-1]
    dirname = os.path.dirname(os.path.realpath(__file__))
    path = os.path.join(dirname, "event_filters", filter_name + ".py")
    if not os.path.exists(path):
        raise SourceFilterNotFoundException(f"Could not find source plugin: {name}")
    return path
```

Because if the rulebook contains a wrong "eda.builtin.missing" filter you want to fail at that point. 

And then you can just do:

```python
            elif has_builtin_filter(source_filter.filter_name):
                source_filter_module = runpy.run_path(
                    get_builtin_filter_path(source_filter.filter_name)
                )
```
But this is only a suggestion, I won't to block the PR.

Reply:
@Alex-Izquierdo I don't think that solves the problem. The has_builtin_filter has to also check if the file exists so that we can continue the search to the next place where the filter can be found. If you just check the name and raise an exception when the file is missing, you cannot continue to looking for the filter in other places.

The has_builtin_filter allows you to check 
1. Does the name start with eda.builtin?
2. Does the file exist?

If both those tests pass we know we have found the filter. If not we can continue looking for the filter in other places.